esp-hal/src/spi/master.rs

```diff
@@ -100,6 +96,43 @@ pub enum SpiInterrupt {
     /// This interrupt is triggered when an SPI transaction has finished
     /// transmitting and receiving data.
     TransDone,
+
+    /// Triggered when the length of GDMA RX FIFO is shorter than that of actual
+    /// data transferred.
+    #[cfg(gdma)]
+    DmaInfifoFullErr,
+
+    /// Triggered when the length of GDMA TX FIFO is shorter than that of actual
+    /// data transferred.
+    #[cfg(gdma)]
+    DmaOutfifoEmptyErr,
+
+    /// Triggered at the end of End_SEG_TRANS transfer in GP-SPI2 slave
+    /// segmented transfer mode or at the end of configurable segmented transfer
+    /// as master.
+    #[cfg(any(esp32s2, gdma))]
+    DmaSegTransDone,
+
+    /// Triggered when a Magic error occurs in CONF buffer during configurable
+    /// segmented transfer as master.
+    #[cfg(any(esp32s2, gdma))]
+    SegMagicErr,
+
+    /// Triggered by RX AFIFO write-full error in GP-SPI2 as master.
+    #[cfg(gdma)]
+    RxAfifoWfullErr,
+
+    /// Triggered by TX AFIFO read-empty error in GP-SPI2 as master.
+    #[cfg(gdma)]
+    TxAfifoRemptyErr,
+
+    /// Used and triggered by software. Only used for user defined function.
+    #[cfg(gdma)]
+    App2,
+
+    /// Used and triggered by software. Only used for user defined function.
+    #[cfg(gdma)]
+    App1,
```

Comment:
There should probably be a way to set these. Not sure if that's in scope for this PR.

Reply:
I agree having a way to listen for them while there is no way to trigger them isn't super useful for now but that's not in scope of this PR